AK/Math.h

```diff
@@ -728,6 +728,30 @@ ALWAYS_INLINE I round_to(double value)
 }
 #endif
 
+template<FloatingPoint T, Integral I>
+constexpr T upow(T x, I n)
+{
+    T y = 1.0;
+
+    do {
+        if (n & 1)
+            y *= x;
+        x *= x;
+        n >>= 1;
+    } while (n);
+
+    return y;
+}
+
+template<FloatingPoint T, Integral I>
+constexpr T ipow(T x, I n)
+{
+    if (n < 0)
+        return upow(1.0l / x, -n);
```

Comment:
Please cast the `1.0l` to a `T`, otherwise it will always coerse upon to use `long doubles`, which are way slower on x86
(x87 instead of SSE and possible vectorization)


Alternative:
Explicitly mark that call to `upow` as `upow<T>`

Reply:
Thanks for the feedback. I see how coercing to `long doubles`  will lead to problems on x86, unfortunately either of the solutions will lead to a failing test. On my x86_64 machine, that `long double` will be coerced to a double (i.e., T) and will fail to provide enough precision for reciprocals. How should we proceed? One possible solution is to leave as it is, add a FIXME mentioning the current issue in the implementation.

Reply:
Googleing around a bit I find
> Originally, the pow(T, int) overloads unrolled the pow call into a sequence of inline multiplication operations. While it was faster, it was also much less accurate. This implementation was removed in Visual Studio 2015 Update 1. For more information, see Conformance improvements in Visual Studio 2015 Update 1.

From https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/pow-powf-powl?view=msvc-170

So it seems that this approach might only be good for "small" exponents anyway